dependency_utils.py

```diff
@@ -134,7 +134,7 @@ def _git_clone(self, context):
         context.force_native_build = True
         if os.path.exists(self.git_path):
             raise SkipCommand()
-        command = "git clone " + self.git_remote
+        command = "git clone {} {}".format(self.git_remote, self.git_dir)
```

Comment:
I was about to tell you it was necessary to get the full history to be able to commit and push updates but it is no more the case \o/

Recent version of git 1.9 (available in trusty) allow to commit and push from shallow repository.
And more recent version (2.8) even allow to unshallow a repository (it is not available in trusty... but I don't care :) )

I will change this, but not in this PR, this is not relevant to the arm functionality (the commit itself is not relevant to arm, so I will remove it from the PR and create another one)